### PR TITLE
PP-12421: Use resource classes

### PR DESCRIPTION
**Pull request by kerrr**
We have added Classes for git and docker images to (pkl-concourse-pipeline)[https://github.com/alphagov/pkl-concourse-pipeline/blob/main/src/Resources.pkl]. Here we use the classes, first by extending them to create Pay specific git and docker hub resource classes and then using those to define the individual resources required.

This PR modifies
* shared_resources.pkl to demonstrate how the exiting functions could be changed to use the new resources.
* concourse-runner.pkl to demonstrate using the resources directly in the pipeline.

The first commit updates the refs to the new version of pkl-concourse-pipeline.

### CODE REVIEW

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-deploy**

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-deploy**

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

File: ci/pkl-pipelines/pay-deploy/concourse-runner.pkl [hunk 1]
```diff
@@ -12,26 +13,78 @@ resource_types {
...
-  shared_resources.payGithubResourceWithBranch("concourse-runner-src", "pay-ci", "master")
-    |> withPath("ci/docker/concourse-runner/*")
-  shared_resources.payGithubResourceWithBranch("concourse-runner-with-java-17-src", "pay-ci", "master")
-    |> withPath("ci/docker/concourse-runner-with-java-17/*")
-  shared_resources.payGithubResourceWithBranch("concourse-runner-with-java-21-src", "pay-ci", "master")
-  |> withPath("ci/docker/concourse-runner-with-java-21/*")
-  shared_resources.payDockerHubResource("concourse-runner", "governmentdigitalservice/pay-concourse-runner", "latest")
-  shared_resources.payDockerHubResource("concourse-runner-with-java-17", "governmentdigitalservice/pay-concourse-runner-with-java-17", "latest")
-  shared_resources.payDockerHubResource("concourse-runner-with-java-21", "governmentdigitalservice/pay-concourse-runner-with-java-21", "latest")
+  new PayResources.PayGitHubPullRequestResource {
```
Comment (kbottla) [inline ci/pkl-pipelines/pay-deploy/concourse-runner.pkl:16]: I think using functions here will reduce the boilerplate code `new ...{}` and keep the module small.

**Review by kbottla — APPROVED** [review]

File: ci/pkl-pipelines/pay-deploy/concourse-runner.pkl (hunk 1 above)
Comment (kerrr) [reply ci/pkl-pipelines/pay-deploy/concourse-runner.pkl:16]: Yeah, was wanting to demo how this worked so we can chat in the PKL retro.

File: ci/pkl-pipelines/pay-deploy/concourse-runner.pkl [hunk 2]
```diff
@@ -12,26 +13,78 @@ resource_types {
...
+    paths {
+      "ci/docker/concourse-runner/**"
+      "ci/docker/concourse-runner-with-java-17/**"
+      "ci/docker/concourse-runner-with-java-21/**"
+      "ci/pipelines/concourse-runner.yml"
+    }
+  }
+  new PayResources.PayGitHubResource {
+    name = "adminusers-master"
+    repoName = "pay-adminusers"
```
[the remark this answers is not in the document]
Comment (kerrr) [reply ci/pkl-pipelines/pay-deploy/concourse-runner.pkl:28]: 👍🏼

**Pull request merged**